user-guide/Advanced_Functionality/Databases/Configuring_dedicated_clustered_storage/Indexing_database/OpenSearch_Database/Installing_OpenSearch_database.md

```diff
@@ -21,13 +24,342 @@ See the [official documentation](https://opensearch.org/docs/latest/) on how to
+
+#### Main steps
+
+These are the main steps of the setup:
+
+1. Install OpenSearch as detailed under [Install OpenSearch from an APT repository](https://opensearch.org/docs/latest/install-and-configure/install-opensearch/debian/#install-opensearch-from-an-apt-repository).
+
+1. Follow [Step 2:(Optional) Test OpenSearch](https://opensearch.org/docs/latest/install-and-configure/install-opensearch/debian/#step-2-optional-test-opensearch) from the installation guide. Remember to also **query the plugins endpoint**.
+
+1. Follow [Step 3: Set up OpenSearch in your environment](https://opensearch.org/docs/latest/install-and-configure/install-opensearch/debian/#step-3-set-up-opensearch-in-your-environment) and change the *JVM Heap Space*.
+
+1. Set up a cluster as detailed under [Creating a cluster](https://opensearch.org/docs/latest/tuning-your-cluster/cluster/). Take the sections below into account when you do so.
+
+#### OpenSearch.yml configuration
+
+- For every node in your cluster, configure the *OpenSearch.yml* file as illustrated in the example below (the example uses three nodes):
+
+  ```yml
+  # Use a descriptive name for your cluster:
+  #
+  cluster.name: NameOfYourCluster
+  cluster.initial_master_nodes: ["opensearchnode1"]
+  #
+  # ------------------------------------ Node ------------------------------------
+  #
+  # Use a descriptive name for the node(node.name can be anything you desire, but we recommend hostname of node):
+  #
+  # Make the node.name reflect the name of your node
+  node.name: opensearchnode1
+  #
+  # ----------------------------------- Paths ------------------------------------
+  #
+  # Path to directory where to store the data (separate multiple locations by comma):
+  #
+  path.data: /var/lib/opensearch
+  #
+  # Path to log files:
+  #
+  path.logs: /var/log/opensearch
+  # ---------------------------------- Network -----------------------------------
+  #
+  # Set the bind address to a specific IP (IPv4 or IPv6), best is to use the real-ip of the node:
+  #
+  network.host: 166.206.186.146
+  
+  network.publish_host: 166.206.186.146
+  #
+  # Set a custom port for HTTP:
+  #
+  http.port: 9200
+  #
+  # For more information, consult the network module documentation.
+  #
+  # --------------------------------- Discovery ----------------------------------
+  #
+  # Pass an initial list of hosts to perform discovery when this node is started:
+  # The default list of hosts is ["127.0.0.1", "[::1]"]
+   #
+  discovery.seed_hosts: ["166.206.186.146","166.206.186.147","166.206.186.148"]
+  
+  discovery.type: zen
+  ```
+
+- If you want a node to be a **data node**, add the following configuration in *OpenSearch.yml*:
+
+  ```yml
+  node.roles: [ data, ingest ]
+  ```
+
+- If you want a node to be the **cluster manager node** (a.k.a. the master node), add the following configuration in *OpenSearch.yml*:
+
+  ```yml
+  node.roles: [ cluster_manager ]
+  ```
+
+##### Remove the demo certificates
```

Comment:
The purpose of this section and the section below it (so line 107 to 121) isn't entirely clear to me yet. Is this something you always need to do, regardless of whether you configure TLS? If yes, could you clarify the purpose of these procedures? If no, and it's actually part of the TLS config, can you move this and include it in the TLS procedure?